## Mini Project on Machine Learning

In this project we are supposed to classify the last variable using Random Forest Classifier
#### Pre Requists:
1. Data set (.csv format)
2. Import necessary libraries

## What is Random Forest?
1. Random forest is a supervised learning algorithm.
2. Used to solve regression and classification problems.
3. The outcome based on the predictions of the decision trees.
4. It uses bagging and feature randomness.
5. Prediction is more accurate than that of any individual tree.

In [1]:
#importing nescessary libraries
import pandas as pd
import numpy as np
from matplotlib import pyplot as plp

In [16]:
data_frame=pd.read_csv("glass.csv")
data_frame.head()

1  1.52101  13.64  4.49   1.1  71.78  0.06  8.75    0   0.1  1.1.1
0  2  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.0  0.00      1
1  3  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.0  0.00      1
2  4  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.0  0.00      1
3  5  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.0  0.00      1
4  6  1.51596  12.79  3.61  1.62  72.97  0.64  8.07  0.0  0.26      1

In [14]:
#Analysing the data
data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 213 entries, 0 to 212
Data columns (total 11 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   1        213 non-null    int64  
 1   1.52101  213 non-null    float64
 2   13.64    213 non-null    float64
 3   4.49     213 non-null    float64
 4   1.1      213 non-null    float64
 5   71.78    213 non-null    float64
 6   0.06     213 non-null    float64
 7   8.75     213 non-null    float64
 8   0        213 non-null    float64
 9   0.1      213 non-null    float64
 10  1.1.1    213 non-null    int64  
dtypes: float64(9), int64(2)
memory usage: 18.4 KB


In [17]:
data_frame.isnull().sum()

1          0
1.52101    0
13.64      0
4.49       0
1.1        0
71.78      0
0.06       0
8.75       0
0          0
0.1        0
1.1.1      0
dtype: int64

In [19]:
data_frame.shape

(213, 11)

## Analysis
1. No  null values are there in the tabular data
2. 9 rows of float values and 2 rows of int values
3. 213 rows and 11 columns